std/asys/Process.hx

```diff
@@ -0,0 +1,178 @@
+package asys;
+
+import haxe.Error;
+import haxe.NoData;
+import haxe.async.*;
+import haxe.io.*;
+import asys.net.Socket;
+import asys.io.*;
+import asys.uv.UVProcessSpawnFlags;
+
+/**
+	Options for spawning a process. See `Process.spawn`.
+**/
+typedef ProcessSpawnOptions = {
+	?cwd:FilePath,
+	?env:Map<String, String>,
+	?argv0:String,
+	?stdio:Array<ProcessIO>,
```

Comment:
Let's change this to `{stdin:ProcessIO, stdout:ProcessIO, stderr:ProcessIO}`? Or make an abstract with `@:from`s for object and array.
I don't want to look into API docs to figure out array indices everytime I need to use this field :)

Reply:
We can have an alternative to the array in the arguments, but in the end (when passed to libuv) it does need to be an array. An arbitrary process does not need to have exactly 3 pipes, nor do they need to be readable and writable the same way standard in, out, and error are.

Reply:
It may be an Array under the hood, but public API should be easy to read/write.